index.rst

```diff
@@ -302,107 +305,156 @@ Is this case, we could have something like:
 
-#.  If a CSC receives a ``start`` command with an empty ``configuration`` (see :ref:`renaming proposal <section-renaming>`) attribute, it shall load the values in ``_base.yaml``.
-#.  If a CSC receives a ``start`` command with a ``configuration`` attribute equal to a label in ``_labels.yaml``, it shall load the values in ``_base.yaml`` first and override those values defined in the mapped configuration file.
-#.  The name ``default`` shall not be used for labels.
-#.  All valid configurations shall have a label to be loaded by the CSC.
-    Configurations cannot be loaded by filename.
-#.  The configuration repository should be kept clean of configurations used for unit testing or for different purposes (e.g. test stand configurations).
-    Configurations tailored for test stand can be kept in different branches/tags.
+#.  If a CSC receives a ``start`` command with an empty ``configurationOverride`` (see :ref:`renaming proposal <section-renaming>`) attribute, it shall load the values in ``_init.yaml`` then the site-specific file (e.g. ``_summit.yaml``).
+
+#.  If a CSC receives a ``start`` command with a ``configurationOverride`` attribute equal to a valid filename, it shall load the values in ``_init.yaml``, then then the site-specific file (e.g. ``_summit.yaml``), and lastly the override file. An invalid filename will return as a failed command with an appropriate error message saying the file was not readable and no state transition will occur.
+
+#.  The configuration repository shall not contain configurations used for unit testing.
     Configurations needed for unit testing shall be added to the ``test`` directory in the CSC repository and use the override feature in CSCs (see `Salobj documentation <https://ts-salobj.lsst.io>`__).
-#.  All configuration files shall have a header explaining that they are loading basic values from ``_base.yaml``, as shown in the :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` mentioned above.
+
+#.  Override configurations that are site specific should contain the site name as a prefix to the filename (e.g. ``summit_simple_algorithm.yaml``).
+
+#.  All configuration files shall have a header metadata fields explaining that they are loading basic values from ``_init.yaml``, as shown in the :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` mentioned above.
+
+
+Required Unit and Continuous Integration (CI) Testing
+-----------------------------------------------------
+
+Due to the dependence of the configuration files on the defined schema, which are located in different repositories, CI tests are required to ensure there is no breakage when making modifications in either repository.
+
+The following CI tests are required on all configuration repos (e.g. ``ts_config_attcs``):
+
+    #. Verify that location configuration files exist for all sites
```

Comment:
Except these are optional. We now need to verify that the `_<site>.yaml` files name valid sites. We might want to require that if one site-specific file is present then we require the whole set (all or none).

Reply:
So for now let's say they're required for all sites. We can revise this later if we need to.
I'm wondering if there are exceptions to this where we don't need configs for all sites. For example, one could populate the `_init.yaml` file with defaults that would work for tests at ncsa and the base, then only a `_summit.yaml` would be required. 
Personally, I think this would be poor practice but it would be hard to police.